tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -275,10 +276,31 @@ def test_host_details_page(
         session.organization.select(org_name=org.name)
         session.location.select(loc_name=DEFAULT_LOC)
         # Sync insights recommendations
+        timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M')
         session.cloudinsights.sync_hits()
-        result = session.host.host_status(rhel_insights_vm.hostname)
-        assert 'Insights: Reporting' in result
-        assert 'Inventory: Successfully uploaded to your RH cloud inventory' in result
+        wait_for(
+            lambda: rhcloud_sat_host.api.ForemanTask()
+            .search(query={'search': f'Insights full sync and started_at >= "{timestamp}"'})[0]
+            .result
+            == 'success',
+            timeout=400,
+            delay=15,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        # Wait for inventory upload status to be updated
+        for i in range(1, 4):
+            result = session.host.host_status(rhel_insights_vm.hostname)
+            try:
+                assert 'Insights: Reporting' in result
+                assert 'Inventory: Successfully uploaded to your RH cloud inventory' in result
+                break
+            except AssertionError:
+                if i == 3:
+                    raise
+                else:
+                    time.sleep(30)
+                    continue
```

Comment:
I understand why are you going this way, but let's try for a better approach :), @jameerpathan111 thoughts?

Reply:
This was just to test the sync status, which is currently failing, but is unrelated to this PR. Will be fixed later, in separate PR. I'll revert this change later